### DISCUSSION TITLE
Collecting Atomic Information from Gromacs Trajectory

### DISCUSSION THREAD

**Discussion by krb0073** [GrossfieldLab/loos#120] [General]
Hey Guys, 

I am writing a code to compute the contribution of all atoms to the electric field of a given vector, but the simulation I have been handed is from a gromacs run . The gro, xtc, pdb, and dcd files I have made to try to perform the code do not have the charge of the atoms from the topology. Short of writing a psf for the simulation to have the atom information, is there a way to collect charges from the standard files LOOS uses to load in gromacs trajectory . Thanks in advance for any feedback

Kyle

**Comment by lgsmith** [comment]
I think charges would be in the `.tpr`. Have you tried to get them into a PDB from the `gmx dump` via `gmxdump2pdb.pl` (which will be in your path if you have a normal loos install)?

**Comment by agrossfield** [comment]
Never fear, we have a solution. LOOS has a Perl script called `gmxdump2pdb.pl` (I can’t recall if cmake strips the “.pl” when it installs, and I’m away from my computer). You run `gmx dump` and pipe the output into this script, and it makes 2 files: a pdb and a “fake” psf (fake in the sense that it doesn’t have everything namd would need to run, just atoms, bonds, charges, and masses). Use that psf as your system file and you’re good to go. 

In general, if you’re analyzing gromacs sims, we recommend doing this, because it gives you a much more information-rich system file. 

If you go to the LOOS wiki, there’s a whole article on analyzing gromacs sims, which includes this tip and others.

**Reply by lgsmith** [reply]
It does not strip the `.pl`, but yeah that's what I suggested too.

**Reply by krb0073** [reply]
Hey guys, 

Thanks for the feedback,  I honestly did not know the pearl script exsisited but i am glad it was an easy fix. I was able to use that within a python code to find I needed. Thanks again!

Kyle

**Comment by agrossfield** [comment]
Happy to help. I'm going to close this discussion, but you can re-open it there's something else.
